### PR TITLE
ImageTexture.load returns an error code.

### PR DESCRIPTION
**Pull request by PJB3005**
Also documented said error code in `Image.load`.

### CODE REVIEW

**Comment by PJB3005** [comment]
Actually wait on #18821 so I can make these enum refs nicer.

File: doc/classes/Image.xml [hunk 1]
```diff
@@ -330,6 +330,7 @@
 			</argument>
 			<description>
 				Loads an image from file [code]path[/code].
+				Returns 0 ([code]OK[/code]) if the load was successful, otherwise an error code from [@GlobalScope].
```
Comment (akien-mga) [inline doc/classes/Image.xml]: You don't need to mention `0`, only `Returns [code]OK[/code]`. People should check against `OK`, not magic numbers.

Comment (akien-mga) [reply doc/classes/Image.xml]: But basically the whole sentence doesn't say much more than the
```
<return type="int" enum="Error">
</return>
```
in my opinion. If we add it here, we would need to add it to all methods that return `Error` and where we don't already document explicitly what kind of error codes can come out.

Comment (PJB3005) [reply doc/classes/Image.xml]: Yeah now that enums are properly documented by the doc writer this could probably be simplified.

**Comment by akien-mga** [comment]
Cherry-picked for 3.0.7.

**Comment by akien-mga** [comment]
Actually no, removed the cherry-pick as it would break GDNative bindings, which we don't want for the stable branch.

**Pull request merged**